File: android/src/main/kotlin/live/hms/hmssdk_flutter/HMSSpeakerExtension.kt
package live.hms.hmssdk_flutter

import live.hms.video.sdk.models.HMSSpeaker

class HMSSpeakerExtension {
    companion object {
        fun toDictionary(speaker: HMSSpeaker?): HashMap<String, Any?>? {
            val hashMap = HashMap<String, Any?>()
            if (speaker == null)return null
            hashMap.put("audioLevel", speaker.level)
            val hmsTrackMap = HMSTrackExtension.toDictionary(speaker.hmsTrack)
            val hmsPeerMap = HMSPeerExtension.toDictionary(speaker.peer)
            if ((hmsTrackMap == null) || (hmsPeerMap == null)) {
                return null
            }
            hashMap.put("track", hmsTrackMap)
            hashMap.put("peer", hmsPeerMap)
            return hashMap
        }
    }
}

File: ios/Classes/Models/HMSSpeakerExtension.swift
//
//  HMSSpeakerExtension.swift
//  hmssdk_flutter
//
//  Created by Yogesh Singh on 21/10/21.
//

import Foundation
import HMSSDK

class HMSSpeakerExtension {
    static func toDictionary(_ speaker: HMSSpeaker) -> [String: Any] {
        let dict = [
            "peer": HMSPeerExtension.toDictionary(peer: speaker.peer),
            "track": HMSTrackExtension.toDictionary(track: speaker.track),
            "audioLevel": speaker.level
        ] as [String: Any]
        
        return dict
    }
}
